### ISSUE TITLE
NAV-FACT- Facturación de derechos de enganche milenial

### ISSUE THREAD

**Issue by bpesquero** [PKFEric/webProxima#1060]
En el contrato  HECEN/13323 se han facturado inoportunamente los derechos de enganche, ya que se trata de un paso a milenial a través de un M. ¿Podríamos revisar la casuística para que no vuelva a suceder?

**Comment by JMunoz17** [comment]
El contrato asociado a la milenial es el HECEN/13383 no? No he visto la factura de compra generada, entiendo que la habéis eliminado no? O podría ser que lo hayas confundido con el contrato antiguo (HECEN/13323) que solo tiene una factura de venta posterior a la de derechos de enganche.

**Comment by bpesquero** [comment]
Perdón, efectivamente es el HECEN/13323
Lo siento

**Comment by JMunoz17** [comment]
Entonces esta todo ok verdad?

**Comment by bpesquero** [comment]
No, sigo sin entender por qué este contrato, el HECEN/13323, tiene una factura de derechos de enganche

**Comment by JMunoz17** [comment]
Pero ese contrato no tiene una tarifa MILENIAL asociada, entiendo que es correcto que tenga su factura de enganche correspondiente. 

El contrato con tarifa milenial es el otro, HECEN/13383.

**Comment by bpesquero** [comment]
La cuestión es que el CUPS ES0031103540831028QZ se le ha cobrado unos derechos de enganche que no le corresponde. Es un cliente que pasó a tarifa milenial y eso implicó un cambio de contrato, pero no entiendo por qué se le facturan lo derechos de enganche.

**Comment by JMunoz17** [comment]
Ok, que no me estaba enterando..

Pues entiendo que ha tenido que ser porque se importó el F1 antes de procesar el M1_05 que es cuando se crea el nuevo contrato.
De tal forma que los datos de F1 se asociaron al contrato antiguo en lugar de al nuevo.

La fecha/hora de creación indican esto:
Peajes de ese F1 -> 26/01/21 12:48
M1_05 -> 26/01/21 18:45

Como hacemos para controlarlo?
Un aviso/error en el log cuando recibamos un F1 de Otras Facturas y tengamos asociado a ese contrato algún M1_02 aceptado? Y que este tenga asociado un M1_01 con cambio de tarifa.

**Comment by bpesquero** [comment]
Me parece bien, pero añadiría en el control que sea de paso a milenial

**Comment by JMunoz17** [comment]
Pero aunque no fuese un paso a milenial ese F1 de derechos de enganche no debería estar asociado al nuevo contrato en lugar de al antiguo?

Pongo aviso en el log o error y que no cargue el F1?

**Comment by bpesquero** [comment]
Idealmente debería ir asociado al nuevo, efectivamente. Lo que más me preocupa es que no se factura la venta.
Cargarlo hay que cargarlo.

**Comment by JMunoz17** [comment]
Subido a PRE y PROD.

Lo que hacemos es cuando viene una factura de derechos de enganche:
- Buscamos el ultimo M1_02 aceptado asociado a ese CUPS.
- Si lo encontramos buscamos el M1_01 asociado a ese M1_02 para ver si tiene cambio de tarifa.
- Buscamos el M1_05 asociado a ese M1_02 para ver si ha sido procesado.

En caso de que encontremos el M1_01 de cambio de tarifa pero no el M1_05 entonces devolvemos error y no permitimos importar el F1 puesto que es necesario procesar el M1_05 primero.
